src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -760,22 +760,25 @@ private UsingDirectiveSyntax ParseUsingDirective()
                 //NB: there's no way this could be true for a set of tokens that form a valid 
                 //using directive, so there's no danger in checking the error case first.
 
-                name = WithAdditionalDiagnostics(CreateMissingIdentifierName(), GetExpectedTokenError(SyntaxKind.IdentifierToken, this.CurrentToken.Kind));
+                type = WithAdditionalDiagnostics(CreateMissingIdentifierName(), GetExpectedTokenError(SyntaxKind.IdentifierToken, this.CurrentToken.Kind));
                 semicolon = SyntaxFactory.MissingToken(SyntaxKind.SemicolonToken);
             }
             else
             {
-                name = this.ParseQualifiedName();
-                if (name.IsMissing && this.PeekToken(1).Kind == SyntaxKind.SemicolonToken)
+                type = alias == null ? this.ParseQualifiedName() : this.ParseType();
+                if (type is not NameSyntax)
+                    type = CheckFeatureAvailability(type, MessageID.IDS_FeatureUsingTypeAlias);
```

Comment:
note: we could `ParseType` universally, but then give a specific error if it's not a named type for normal `using directives`.  However, the upside of this appraoch is that someone can always assume if they have a `normal using directive` or `using static directive` that there is a name.

Reply:
See https://github.com/dotnet/roslyn/pull/50167/files#diff-95477edd7a2f9bf55ac156fbeb22d71f973b06dc20a7034b373e8c2e8341d6f0R11 for the reasoning here.